File: golang-alg/sort/selectionSort.go
package main

import (
	"fmt"
)

func main() {
	arr := []int{1, 9, 2, 10, 30, 5, 45, 8}
	fmt.Println(selectionSort(arr))
}

// ===
// 选择排序
// ===
func selectionSort(arr []int) []int {
	if len(arr) <= 1 {
		return arr
	}
	for i := 0; i < len(arr)-1; i++ {
		minIndex := i
		for j := i + 1; j < len(arr); j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}
		arr[i], arr[minIndex] = arr[minIndex], arr[i]
	}
	return arr
}

File: python-alg/basic/selection_sort.py
def selection_sort(arr):
    for i in range(len(arr)):
        smallest_index = i
        for j in range(smallest_index + 1, len(arr)):
            if arr[j] < arr[smallest_index]:
                smallest_index = j

        if smallest_index != i:
            temp = arr[i]
            arr[i] = arr[smallest_index]
            arr[smallest_index] = temp


my_list = [64, 34, 25, 12, 22, 11, 90]
selection_sort(my_list)
print(my_list)
